src/Illuminate/Database/Connectors/MySqlConnector.php

```diff
@@ -104,24 +104,51 @@ protected function getHostDsn(array $config)
     }
 
     /**
+     * Get the modes for the connection.
+     *
+     * @param  array  $config
+     * @return array
+     */
+    protected function getModes(array $config)
+    {
+        if (isset($config['modes'])) {
+            return (array) $config['modes'];
+        }
+
+        if (! isset($config['strict'])) {
+            return [];
+        }
+
+        if (! $config['strict']) {
+            return ['NO_ENGINE_SUBSTITUTION'];
+        }
```

Comment:
I wonder if two specific conditions for `$config['strict'] === false` and `$config['strict'] === true` would be clearer to come back and understand vs. relying on truthiness? And maybe throw an `InvalidArgumentException` as the default case?